### ISSUE TITLE
🐛 Resource tracking cannot be turned off

### ISSUE THREAD

**Issue by cgood92** [DataDog/browser-sdk#2143]
**Describe the bug**
This is a problem with DataDog RUM.

While using `@datadog/browser-rum-slim`, v4.30.0 and v4.38.0 (latest), I have this configuration:

```
const ddRumConfig = {
  clientToken: 'SCRUBBED',
  applicationId: 'SCRUBBED',
  version: process.env.VERSION,
  sampleRate: 100,
  sessionReplaySampleRate: 0,
  trackUserInteractions: false,
  trackInteractions: false,
  trackResources: false,
  env: 'SCRUBBED',
};

    <script src="datadog-v4.38.0" type="text/javascript"></script>
    <script>
        window.DD_RUM && DD_RUM.onReady(function() {
            window.DD_RUM.init(ddRumConfig);
        })
    </script>
```

But, even with that, resources are still being tracked.

The expectation is that resources (ajax, css, js, etc) would not be tracked and sent to DataDog when the `trackResources: false` is set in the configuration.

I cloned your repo, and began looking into the problem, and here are some of my notes:
1. If you search for `trackResources`, you will see that it is not used directly in any logic or conditions, but it is mapped to `resourceAllowed`.
2. If you search for `resourceAllowed`, I can see that it is only used in `computeIndexingInfo` to set `discarded` as true.  But, that doesn't stop the request from going through.

So, it looks to me as though the setting `trackResources` is not actually doing anything.

In addition to that, I think if someone sets `trackResources` to be `false`, that you should never run `startResourceCollection` or `startRequestCollection`, because:
1. If we're not going to track these resources in DataDog, then don't spend cycles instrumenting and tracking them (this just slows down our system).
2. Instrumenting them ruins stack traces for no reason if we're not going to use DataDog.

**Comment by BenoitZugmeyer** [comment]
Thank you for your report.

Could you tell me whether resources are showing up in the Datadog App?

I understand your concerns about resources sent to Datadog even if resources tracking is disabled. It works as designed: the resources should be droped when being processed by the backend. We'll see if we can improve this.

**Comment by cgood92** [comment]
> Thank you for your report.
> 
> Could you tell me whether resources are showing up in the Datadog App?
> 
> I understand your concerns about resources sent to Datadog even if resources tracking is disabled. It works as designed: the resources should be droped when being processed by the backend. We'll see if we can improve this.

Thank you @BenoitZugmeyer.  These resources are no longer showing up in the DataDog App.

As you pointed out, my concern is about the noisiness.  On a single page load for my application, there are 69 network requests sent to DataDog reporting on resources.  Very noisy.  Plus, we have other instrumentation implementations too, and each time they send a request to report to their event hub DataDog then tries to report on those as well.  So, very, very noisy.

The longer history is that we used to have resource tracking just working for us automatically.  Then, it stopped a few months ago for no reason.  Then, we passed along `trackResources: true` to get it back, and they started to come again.

Well, now we're trying to unwind that and NOT track resources.  Way too noisy.

For now, I have implemented this in our beforeSend:

```
if (event.type === 'resource') {
  return false;
}
```

But, it would be nice if the flag was honored instead of having to do this.  Plus, as mentioned, it would be nice if DataDog just didn't instrument things if we're asking it not to, because of the useless time it spends tracking timings and all of that jazz.

Thanks again for taking a look.  We're heavy DataDog users.
